### PR TITLE
Add testsuite for unit- and integration tests

### PR DESCRIPTION
**Pull request by gribok**
Based on #64, I created a basic test suite with following features:
* Unit-Testing of all help function, like `date.is_date`
* Integration-Testing of API endpoints (validation agains output from `tests/expected_output`)
* Add example data to `tests/example_data` for testing
* Mocking of `app.data.requests.get` and `app.data.datetime`  to enforce data fetch locally from `tests/example_data`
* Edit of `README.md` for testing and linting
* Add `pytest` and `pytest` to `Pipfile`  to `dev-packages`
* Randomize testing possible to test for stable code
* Update `Pipfile.lock` with new dev-dependencies

Testing of PR:
``` bash
$ pytest -s -v tests/
```

ToDos:
* Add additonal testcases for endpoint `/all`. It is hard to test due to caching mechanism. Need to check, if it is possible to deactivate caching during testing
* Optional: Contol testing per `Makefile`
* Optional: Add to CI-pipeline

### CODE REVIEW

**Comment by ExpDev07** [comment]
This is gonna conflict with /v2/: https://github.com/ExpDev07/coronavirus-tracker-api/tree/v2 😢

**Comment by gribok** [comment]
No problem. I will waiting for release of v2 and adapt the testcases

**Comment by ExpDev07** [comment]
@gribok Really appreciate it.

**Comment by ExpDev07** [comment]
You should be good to make your edits now :). /v2/ is released.

File: tests/test_data.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from unittest import mock
+import pytest
+
+from app import data
```
Comment (ExpDev07) [inline tests/test_data.py]: The data module no longer exists.

**Comment by ExpDev07** [comment]
Conflicts needed to be resolved.

**Comment by gribok** [comment]
Fixed all merge conflicts.
Adapt testsuite to v2.

Check it out. :-)

Some points to highlights;
* Due to cache mechanism, it is important to mock same datetime for `latest` key
   * Create constant variable `test_jgu.DATETIME_STRING`
* Return data of API endpoints validate against expected output for /v1 and /v2 endpoints

Would be nice, to be added to contrib-list

File: tests/test_data.py (hunk 1 above)
Comment (gribok) [reply tests/test_data.py]: Already obsolet. Accidentally push commits to early. :-)

**Comment by ExpDev07** [comment]
Looks good sir.

**Comment by ExpDev07** [comment]
@allcontributorsbot please add @gribok  for code and testing.

**Comment by ExpDev07** [comment]
@AllContributorsBot please add @gribok for code and testing.

**Comment by ExpDev07** [comment]
@all-contributors please add @gribok for code and testing.

**Pull request merged**